Project-Service-implementation-Script/main.py

```diff
@@ -4196,6 +4202,121 @@ def solutionCreationAndMapping(projectName_for_folder_path, entityToUpload, list
             print("Project solution creation api failed.")
             sys.exit()
 
+
+# Function to download evidence files dynamically
+def downloadEvidences(filePathAddProject, projectName_for_folder_path):
+    # Open the workbook and find the sheet
+    wbproject = xlrd.open_workbook(filePathAddProject, on_demand=True)
+    projectSheet = wbproject.sheet_names()
+    for prosheet in projectSheet:
+        if prosheet.strip().lower() == 'Project upload'.lower():
+            print("--->Checking Project details sheet...")
+            detailsEnvSheet = wbproject.sheet_by_name(prosheet)
+            # Read the header row to identify evidence columns
+            keysEnv = [detailsEnvSheet.cell(1, col_index_env).value for col_index_env in range(detailsEnvSheet.ncols)]
+            
+            # Create a folder to store evidence files
+            Logofilepath = os.path.join(projectName_for_folder_path, 'evidenceFile')
+            if not os.path.exists(Logofilepath):
+                os.mkdir(Logofilepath)
+            
+            # Loop through rows starting from the 2nd data row (row index 2)
+            for row_index_env in range(2, detailsEnvSheet.nrows):
+                dictDetailsEnv = {
+                    keysEnv[col_index_env]: detailsEnvSheet.cell(row_index_env, col_index_env).value
+                    for col_index_env in range(detailsEnvSheet.ncols)
+                }
+                
+                # Process all columns with "evidence link" in their header
+                for key, value in dictDetailsEnv.items():
+                    if "evidence link" in key.lower() and value:
+                        value = dictDetailsEnv.get(key, "").strip()
+                        link_type_key = key.replace("link", "type")  # Dynamically match the type column
+                        link_type = dictDetailsEnv.get(link_type_key, "").strip().lower()
+                        if value and link_type:
+                            try:
+                                logo_split = str(value).split('/')[5]  # Extract file ID from Google Drive link
+                                file_url = f'https://drive.google.com/uc?export=download&id={logo_split}'
```

Comment:
@MallanagoudaB are we expecting google cloud link for every situation? what if user provide any other download link?